cep-purls.md

```diff
@@ -0,0 +1,192 @@
+
+### PURLs in recipes
+
+PURLs can be specified in the recipe under `.about.purls` (implementation for [v1-recipes](cep-00014.md): [prefix-dev/rattler-build #1664](https://github.com/prefix-dev/rattler-build/pull/1664)):
+
+```yaml
+about:
+  # ...
+  # PURLs that this package belongs to
+  purls: [string (PURL enforced)]
+```
+
+Tools that build conda packages should add these packages to `index.json`:
+
+```json
+{
+  ...,
+  "purls": ["pkg:pypi/pinject@0.14.1"]
+}
+```
+
+### PURLs in repodata
+
+In `repodata.json`, PURLs should be an optional field.
+Tools that generate `repodata.json` (like `rattler-index` or `conda-index`) can decide whether to include it or not in `repodata.json`.
+
+Instead of writing the PURLs in `repodata.json`, these tools can decide to add PURLs to a separate `purls.json` like in [CEP 12](cep-0012.md):
+
+```json
+{
+  "info": {
+    "platform": "string",
+    "arch": "string",
+    "subdir": "string",
+    "version": 0
+  },
+  "packages": {
+    "package-version-build.tar.bz2": {
+      ...,
+      "purls": ["pkg:pypi/pinject@0.14.1"]
+    }
+  },
+  "packages.conda": {
+    "package-version-build.conda": {
+      ...,
+      "purls": ["pkg:pypi/pinject@0.14.1"]
+    }
+  }
+}
+```
+
+If the `purls` field is not present in the `repodata.json` record, it means no `purls` information is stored with the record. Then, a fallback mechanism (i.e., read `purls.json`) should be used to acquire the PURL information.
+
+In sharded repodata ([CEP 16](cep-0016.md)), PURLs can be specified similar to [CEP 21](cep-0021.md).
+
+### Patching
+
+With the purls part of the repodata we propose to also allow repodata patching this field. The original purls can still be extracted from the packages in `index.json`. We do not see a reason how the PURL information is different from other patchable information stored in the repodata (like the dependencies).
+
+To facilitate these patches, the `patch_instructions_version` in `patch_instructions.json` files is incremented to `3`. A `patch_instructions_version: 3` file MAY contain a `purls: ...` field in the patch instructions that MUST be used to patch generated the `purls` entry in the (sharded) repodata (or the `purls.json`).
+
+```json
+{
+  "patch_instructions_version": 3,
+  "packages": {
+    "package-version-build.tar.bz2": {
+      ...,
+      "purls": ["pkg:pypi/pinject@0.14.1"]
+    },
+    ...
+  }
+}
+```
+
+### PURL of a conda packages
+
+Conda packages itself should have a PURL as well.
+This makes it possible for CVE Numbering Autorities (CNA) to publish vulnerabilities for conda packages.
+A package `pinject-0.14.1-pyh29332c3_1.conda` published on `conda-forge` should have the PURL `pkg:conda/conda-forge/pinject@0.14.1?build=pyh29332c3_1&platform=noarch`.
+Packages can also specify a custom `repository_url`, for example `pkg:conda/custom-channel/my-package@0.1.0?build=pyh29332c3_1&platform=noarch&repository_url=prefix.dev`.
```

Comment:
If I'm not mistaken, `repository_url` should be an actual URL, not a domain name: https://github.com/package-url/purl-spec/blob/main/PURL-SPECIFICATION.rst#known-qualifiers-keyvalue-pairs

>repository_url is an extra URL for an alternative, non-default package repository or registry. When a package does not come from the default public package repository for its type a purl may be qualified with this extra URL. The default repository or registry of a type is documented in the "Known purl types" section.